selftests/checkall

```diff
@@ -36,17 +36,47 @@ parallel_selftests() {
         python -m unittest ${ALL[@]:$I:$PER_SLICE} &> $TMP &
         PIDS+=("$!")
     done
+    FAILED_ONCE=()
     for I in $(seq 0 $((${#PIDS[@]} - 1))); do
         wait ${PIDS[$I]}
         RET=$?
         if [ $RET -ne 0 ]; then
+            for FAILURE in $(cat "${TMPS[$I]}" | sed -n 's/\(ERROR\|FAIL\): \([^ ]*\) (\([^)]*\)).*/\3.\2/p'); do
+                FAILED_ONCE+=("$FAILURE")
+            done
+        else
+            rm ${TMPS[$I]}
+        fi
+    done
+    if [ ${#FAILED_ONCE[@]} -gt 0 ]; then
+        if [ ${#FAILED_ONCE[@]} -le 10 ]; then
```

Comment:
Ok, not sure where this "10" magic number came from, but it serves to its purpose.

Reply:
It's meant to avoid re-running jobs in case most of the tests fail. Anyway as I used `--fail-fast` it might not be necessary... We'll see... (ps. It was based on fact that 1-2 tests usually fail due to missing file issue so 10 seemed safe enough, but in Travis more tests are failing due to computation power, so we might need to increase it in the future)